OpenRA.Mods.Common/Widgets/ViewportControllerWidget.cs

```diff
@@ -305,6 +310,66 @@ public override bool HandleKeyPress(KeyInput e)
 				return true;
 			}
 
+			if (key == ks.ViewPortBookmarkSaveSlot1)
+			{
+				viewPortBookmarkSlot1 = worldRenderer.Viewport.CenterPosition;
+				return false;
+			}
+
+			if (key == ks.ViewPortBookmarkSaveSlot2)
+			{
+				viewPortBookmarkSlot2 = worldRenderer.Viewport.CenterPosition;
+				return false;
+			}
+
+			if (key == ks.ViewPortBookmarkSaveSlot3)
+			{
+				viewPortBookmarkSlot3 = worldRenderer.Viewport.CenterPosition;
+				return false;
+			}
+
+			if (key == ks.ViewPortBookmarkSaveSlot4)
+			{
+				viewPortBookmarkSlot4 = worldRenderer.Viewport.CenterPosition;
+				return false;
+			}
+
+			if (key == ks.ViewPortBookmarkUseSlot1)
+			{
+				if (viewPortBookmarkSlot1.X != 0)
+				{
```

Comment:
Please note that our [code style](https://github.com/OpenRA/OpenRA/wiki/Coding-Standard) is to omit braces around single line blocks.